lib/azure/armrest/template_deployment_service.rb

```diff
@@ -49,6 +49,68 @@ def get_template(deploy_name, resource_group = configuration.resource_group)
         response = JSON.parse(rest_post(url))['template']
         DeploymentTemplate.new(response)
       end
+
+      # Delete a deployment and all associated resources that were generated by the
+      # deployment.
+      #
+      # This is a synchronous call. It waits until all deletions complete
+      def delete_associated_resources(deploy_name, resource_group = configuration.resource_group)
+        operations = list_deployment_operations(deploy_name, resource_group)
+        resource_urls = operations.collect do |op|
+          if op.properties.provisioning_operation == 'Create'
```

Comment:
I'm super paranoid about Azure changing the case, as I have seen both upper and lower case responses, depending on the situation. I would recommend using casecmp('Create') instead.